src/main/groovy/io/micronaut/build/MicronautBuildCommonPlugin.groovy

```diff
@@ -108,6 +108,37 @@ class MicronautBuildCommonPlugin implements Plugin<Project> {
             }
         }
 
+        // Predictive test selection is enabled if:
+        // an environment variable is explicitly configured and set to true
+        // or a system property is explicitly configured and set to true
+        // or it's a local build
+        def testSelectionEnabled = project.providers.environmentVariable("PREDICTIVE_TEST_SELECTION")
+                .orElse(project.providers.systemProperty("predictiveTestSelection"))
+                .map {
+                    if (it.trim()) {
+                        Boolean.parseBoolean(it)
+                    } else {
+                        true
+                    }
+                }
+                .orElse(micronautBuildExtension.environment.isNotGithubAction())
+
+        project.dependencies {
+            testImplementation(testSelectionEnabled.map { enabled ->
+                if (enabled) {
+                    platform('org.junit:junit-bom') {
+                        version {
+                            require '[5.8,)'
+                            prefer '5.8.2'
+                        }
+                        because "Predictive test selection requires JUnit 5.8+"
```

Comment:
FWIW, this is required because Micronaut Core doesn't directly depend on JUnit Platform, nor on its BOM, so upgrading to 5.8.2 in the Micronaut BOM isn't enough. Projects which depend on the BOM would require a new version which depends on 5.8+. In the meantime, I'm adding the constraint if test selection is enabled.

Reply:
Probably something like this: https://dev.to/ku6ryo/how-to-set-env-parameters-conditionally-in-github-actions-1g6a